### PR TITLE
797 memory leak on delete

### LINKED ISSUE #797: Memory leak during record deletion
Reproduces with single node cluster with single vdisk (configs from `config-examples`) with the following `bobp` command:
```
bobp -b test -c 500000 --user admin --password password -t 6 
```

Perhaps the bug can be connected with bug in ObserverWorker, that never stops. This bug was fixed in https://github.com/qoollo/pearl/pull/265, thus the test should be done again after the new Pearl will be released

### PR DESCRIPTION
**Pull request by idruzhitskiy**
Closes #797

### CODE REVIEW

File: bob-apps/Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,9 @@ edition = "2021"
 default = ["mimalloc"]
 mimalloc = ["dep:mimalloc"]
 mimalloc-secure = ["dep:mimalloc", "mimalloc/secure"]
+# Important: can't be used with panic = "abort"
+# Please use profile "jemalloc-profile"
+jemalloc-profile = ["jemallocator", "jemallocator/profiling"]
```
Comment (ikopylov) [inline bob-apps/Cargo.toml:23]: Please, use `dep` prefix for dependencies: 
```suggestion
jemalloc-profile = ["dep:jemallocator", "jemallocator/profiling"]
```

Comment (idruzhitskiy) [reply bob-apps/Cargo.toml:23]: It doesn't work with features, "jemallocator/profiling" can't be used with "dep:jemallocator"

Comment (idruzhitskiy) [reply bob-apps/Cargo.toml:23]: I've fixed mimalloc-secure as it was not compiling as well

File: bob-apps/bin/bobd.rs [hunk 2]
```diff
@@ -1,4 +1,9 @@
+#[cfg(all(any(feature = "mimalloc", feature = "mimalloc-secure"), any(feature = "jemallocator", feature = "jemallocator-profile")))]
+compile_error!("features `mimalloc` and `jemalloc-profile` are mutually exclusive");
```
Comment (ikopylov) [inline bob-apps/bin/bobd.rs:2]:
```suggestion
compile_error!("features `mimalloc` and `jemallocator` are mutually exclusive");
```

**Pull request merged**